test/web-auth/captcha.test.js

```diff
@@ -0,0 +1,224 @@
+        templates: {
+          auth0: challenge => `custom template: ${challenge.image}`
+        }
+      });
+    });
+
+    it('should show the element', function () {
+      expect(element.style.display).to.equal('');
+    });
+
+    it('should set the image tag', function () {
+      expect(element.innerHTML).to.equal(`custom template: ${challenges[0].image}`);
+    });
+  });
+
+  describe('when challenge is required and provider is recaptcha', function () {
+
+    const challenge = {
+      required: true,
+      provider: 'recaptcha_v2',
+      siteKey: 'blabla sitekey'
+    };
+
+    let c, recaptchaScript, scriptOnLoadCallback, element;
+
+    beforeEach(() => {
+      const { window } = new JSDOM('<body><div class="captcha" /></body>');
+      element = window.document.querySelector('.captcha');
+      global.window = window;
+      const mockClient = {
+        getChallenge(cb) {
+          cb(null, challenge);
+        }
+      };
+      c = captcha.render(mockClient, element);
+      recaptchaScript = [...window.document.querySelectorAll('script')].find(s => s.src.match('google\.com'));
+      scriptOnLoadCallback = window[url.parse(recaptchaScript.src, true).query.onload];
+    });
+
+    afterEach(function () {
+      delete global.window;
+    });
+
+    it('should inject the recaptcha script', function () {
+      expect(recaptchaScript.async).to.be.ok();
+      const scriptUrl = url.parse(recaptchaScript.src, true);
+      expect(scriptUrl.hostname).to.equal('www.google.com');
+      expect(scriptUrl.pathname).to.equal('/recaptcha/api.js');
+      expect(scriptUrl.query.hl).to.equal('en');
+      expect(scriptUrl.query).to.have.property('onload');
+    });
+
+    describe('after captcha is loaded', function () {
+      let renderOptions;
+      let renderElement;
+      let reseted = false;
+
+      beforeEach(function () {
+        reseted = false;
+        window.grecaptcha = {
+          render(element, options) {
+            renderElement = element;
+            renderOptions = options;
+            return 0;
+          },
+          reset() { reseted = true; }
+        };
+        scriptOnLoadCallback();
+      });
+
+      it('should render with the site key', function () {
+        expect(renderOptions.sitekey).to.equal(challenge.siteKey);
+      });
+
+      it('should set the value on the input when the user completes the captcha', function () {
+        const mockToken = 'token xxxxxx';
+        const input = element.querySelector('input[name="captcha"]');
+        renderOptions.callback(mockToken)
+        expect(input.value).to.equal(mockToken);
+      });
+
+      it('should clean the value when the token expires', function () {
+        const input = element.querySelector('input[name="captcha"]');
+        input.value = 'expired token';
+        renderOptions['expired-callback']()
+        expect(input.value).to.equal('');
+      });
+
+      it('should clean the value when there is an error', function () {
```

Comment:
I think this file isn't covering [this error scenario](https://github.com/auth0/auth0.js/pull/1126/files#diff-a67fdcecd6817c6a8a2f33827850b468R94)